### PR TITLE
added implementation car and driver jdbc, removed storage, car and dr…

### PR DESCRIPTION
**Pull request by Fedich09**
…iver dao, added generic interface

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/mate/dao/jdbc/CarJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setObject(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            addIntoCarsDriversTable(car, connection);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/dao/jdbc/CarJdbcImpl.java]:
```suggestion
            insertCarDrivers(car, connection);
```

File: src/main/java/mate/dao/jdbc/CarJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                + "     AND m.is_delete = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        ResultSet.TYPE_SCROLL_SENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY)) {
+            statement.setObject(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.first();
+            Car car = createNewCar(resultSet);
+            List<Driver> drivers = addAllDriversIntoCar(car.getId(), resultSet);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/dao/jdbc/CarJdbcImpl.java]:
```suggestion
            List<Driver> drivers = getCarDrivers(car.getId(), resultSet);
```

File: src/main/java/mate/dao/jdbc/CarJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                + "WHERE c.is_delete = FALSE"
+                + "     AND d.is_delete = FALSE"
+                + "     AND m.is_delete = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        ResultSet.TYPE_SCROLL_SENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY)) {
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            List<Car> cars = addAllCarsIntoList(resultSet);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/dao/jdbc/CarJdbcImpl.java]:
```suggestion
            List<Car> cars = getCars(resultSet);
```

File: src/main/java/mate/dao/jdbc/CarJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+                + "WHERE"
+                + "    car_id = ? "
+                + "AND "
+                + "     c.is_delete = FALSE ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setObject(2, car.getManufacturer().getId());
+            statement.setObject(3, car.getId());
+            deleteFromCarsDriversTable(car.getId(), connection);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/dao/jdbc/CarJdbcImpl.java]:
```suggestion
            deleteCarDrivers(car.getId(), connection);
```

File: src/main/java/mate/dao/jdbc/CarJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+            throw new DataProcessingException("Can't create new car", e);
+        }
+    }
+
+    private List<Driver> addAllDriversIntoCar(Long id, ResultSet resultSet) {
+        try {
+            resultSet.beforeFirst();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                if (resultSet.getObject("car_id", Long.class).equals(id)) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/dao/jdbc/CarJdbcImpl.java:236]: add this to while condition

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/mate/dao/jdbc/CarJdbcImpl.java (hunk 5 above)
Comment (mfedenko09) [reply src/main/java/mate/dao/jdbc/CarJdbcImpl.java:236]: in this case I can't go through whole Result Set, I add just first car and while loop are ended

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**